hooks/useStakedBalance.ts

```diff
@@ -12,14 +12,14 @@ import { useStakingContract } from './useStakingContract'
 export function useStakedBalance() {
   const { legacyContract, newContract } = useStakingContract(true)
 
-  const account = useRecoilValue(accountState)
+  const { address: account } = useAccount()
```

Comment:
Maybe we should prefer to check the `isConnected` field? https://wagmi.sh/docs/hooks/useAccount#return-value

It seems like quite a lot of `address` references are only there to check if it exists. I'm unsure if there can be no address while being 'connected' though.